common-ts/src/common-ui-utils/settings/settings.ts

```diff
@@ -83,37 +83,39 @@ export class VersionedSettingsHandler<T extends VersionedSettings> {
 
 		const newSettings = { current: settings };
 
-		let transformationWasApplied = false;
+		const transformsApplied: SettingsVersionRule<T>[] = [];
 
 		rulesToApply.forEach((rule) => {
 			newSettings.current = produce(newSettings.current, (draft) => {
-				const { transformationWasApplied: transFormApplied } =
+				const { transformationWasApplied: ruleTransformApplied } =
 					rule.handler(draft);
 
-				if (transFormApplied) {
-					transformationWasApplied = true;
+				if (ruleTransformApplied) {
+					transformsApplied.push(rule);
 				}
 			});
 		});
 
+		const aTransformWasApplied = transformsApplied.length > 0;
+
 		newSettings.current = produce(newSettings.current, (draft) => {
 			draft.version = maxVersion;
 		});
 
 		let transformationDescription = '';
 
-		if (rulesToApply.length === 1) {
-			transformationDescription = rulesToApply[0].longChangeDescription;
+		if (transformsApplied.length === 1) {
+			transformationDescription = transformsApplied[0].longChangeDescription;
 		}
 
-		if (rulesToApply.length > 1) {
-			transformationDescription = `The following settings were changed to support new features : ${rulesToApply
+		if (transformsApplied.length > 1) {
+			transformationDescription = `The following settings were changed to support new features : ${transformsApplied
```

Comment:
this is the crux of these changes : it was applying all of the rule descriptions if any transform was applied, not just the ones which ACTUALLY had to be applied